src/Uno.UI/UI/Xaml/DependencyPropertyDetails.cs

```diff
@@ -34,76 +41,107 @@ static DependencyPropertyDetails()
 			}
 		}
 
-        private DependencyPropertyDetails()
-		{
-			_stack = _pool.Rent(_stackLength);
-		}
+		private List<BindingExpression> Bindings => _bindings ??= new List<BindingExpression>();
+
+		private bool HasBindingsList => _bindings != null;
 
 		public void Dispose()
 		{
 			CallbackManager.Dispose();
-			_pool.Return(_stack, clearArray: true);
+
+			if (_stack != null)
+			{
+				_pool.Return(_stack, clearArray: true);
```

Comment:
```suggestion
				_pool.Return(_stack, clearArray: false);
```
It's useless to clear arrays, since we'll reset them _manuall_ each time we're renting a new one.

Reply:
It's actually required, otherwise it'll be leaking instance stored in the arrays. I'll add it to the code to clarify.